plutus-ledger-constraints/src/Ledger/Constraints/OffChain.hs

```diff
@@ -396,14 +398,13 @@ mkTx lookups txc = mkSomeTx [SomeLookupsAndConstraints lookups txc]
 
 -- | Each transaction output should contain a minimum amount of Ada (this is a
 -- restriction on the real Cardano network).
---
--- TODO: In the future, the minimum Ada value should be configurable.
-adjustUnbalancedTx :: UnbalancedTx -> UnbalancedTx
-adjustUnbalancedTx = over (tx . Tx.outputs . traverse) adjustTxOut
+adjustUnbalancedTx :: ProtocolParameters -> UnbalancedTx -> UnbalancedTx
+adjustUnbalancedTx pparams = over (tx . Tx.outputs . traverse) adjustTxOut
   where
     adjustTxOut :: TxOut -> TxOut
     adjustTxOut txOut =
-      let missingLovelace = max 0 (Ledger.minAdaTxOut - Ada.fromValue (txOutValue txOut))
+      let minAdaTxOut' = either (const minAdaTxOut) (Ada.fromValue . evaluateMinLovelaceOutput pparams) $ fromPlutusTxOut' txOut in
```

Comment:
Not sure here, but feels like we shouldn't default to `minAdaTxOut` because of a conversion error between Plutus TxOut and Cardano TxOut. Someone who calls `adjustUnbalancedTx` would expect it to always work because of the signature. Maybe return `Either ... UnbalancedTx`?

Reply:
Agree, I choose `minAdaTxOut` as a default behaviour. Returning `Either` sounds better.